src/pages/joblist/components/MainFilter.js

```diff
@@ -0,0 +1,298 @@
+import { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import { FaAngleDown } from 'react-icons/fa';
+
+const MainFilter = ({ selectHandler }) => {
+  const [isClickBtn, setIsClickBtn] = useState(false);
+  const [checkMainFilter, setCheckMainFilter] = useState({
+    all: true,
+    fullStack: false,
+    frontEnd: false,
+    backEnd: false,
+  });
+
+  const { all, fullStack, frontEnd, backEnd } = checkMainFilter;
+
+  const buttonHandler = value => {
+    if (value === 'all' && !all) {
+      setCheckMainFilter({
+        all: true,
+        fullStack: false,
+        frontEnd: false,
+        backEnd: false,
+      });
+    }
+
+    if (value !== 'all') {
+      if (!fullStack || !frontEnd || !backEnd) {
+        setCheckMainFilter(prevState => {
+          return {
+            ...prevState,
+            all: false,
+            [value]: !prevState[value],
+          };
+        });
+      } else {
+        setCheckMainFilter(prevState => {
+          return {
+            ...prevState,
+            [value]: !prevState[value],
+          };
+        });
+      }
+    }
+  };
+
+  const objToList = obj => {
+    const result = [];
+
+    for (let key in obj) {
+      if (obj[key]) {
+        if (key === 'all') {
+          return [key];
+        } else {
+          result.push(key);
+        }
+      }
+    }
+
+    return result;
+  };
+
+  const submitHandler = () => {
+    selectHandler('job', objToList(checkMainFilter));
+    clickHandler();
+  };
+
+  const clickHandler = () => {
+    setIsClickBtn(prevState => !prevState);
+  };
+
+  useEffect(() => {
+    if (!fullStack && !frontEnd && !backEnd) {
+      setCheckMainFilter({
+        all: true,
+        fullStack: false,
+        frontEnd: false,
+        backEnd: false,
+      });
+    }
+  }, [fullStack, frontEnd, backEnd]);
+
+  const mainFilterContent = objToList(checkMainFilter)
+    .map(ele => {
+      if (ele === 'all') {
+        return '개발 전체';
+      } else if (ele === 'fullStack') {
+        return '풀스택 개발자';
+      } else if (ele === 'frontEnd') {
+        return '프론트엔드 개발자';
+      } else if (ele === 'backEnd') {
+        return '백엔드 개발자';
+      }
+      return null;
+    })
+    .join(', ');
```

Comment:
객체로 조금 더 깔끔하게 구현할 수도 있을 것 같아요!
```suggestion
  const mainFilterContent = objToList(checkMainFilter)
    .map(stack => {
      const titleList = {
        all: '개발 전체',
        fullStack: '풀스택 개발자',
        frontEnd: '프론트엔드 개발자',
        backEnd: '백엔드 개발자',
      };
      return titleList[stack];
    })
    .join(', ');
```